tests/test_driver.py

```diff
@@ -19,40 +15,33 @@
 
 class TestRelayDriver(unittest.TestCase):
     def test_constructor(self):
-        GPIO_mock.setmode.assert_called_with("MockedBCM")
-        relay = Relay(18, True)
+        relay = Relay.get_or_create_relay(1, True, MockFactory())
```

Comment:
1. this test is for the class constructor, the static method you made should be tested separately
2. please preserve the pin numbers of the tests, only change logic, not data (here and below)

```suggestion
        relay = Relay(18, True, MockFactory())
```

Reply:
fixed